projects/plugins/boost/app/modules/cache/Logger.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Page_Cache;
+
+/**
+ * A utility that manages logging for the boost cache.
+ */
+class Logger {
+	/**
+	 * The singleton instance of the logger.
+	 */
+	private static $instance = null;
+
+	/**
+	 * The header to place on top of every log file.
+	 */
+	const LOG_HEADER = "<?php die(); // This file is not intended to be accessed directly. ?>\n\n";
+
+	/**
+	 * Get the singleton instance of the logger.
+	 */
+	public static function get_instance() {
+		if ( self::$instance !== null ) {
+			return self::$instance;
+		}
+
+		$instance          = new Logger();
+		$prepared_log_file = $instance->prepare_file();
+		if ( is_wp_error( $prepared_log_file ) ) {
+			return $prepared_log_file;
+		}
+
+		self::$instance = $instance;
+		return $instance;
+	}
+
+	/**
+	 * Ensure that the log file exists, and if not, create it.
+	 */
+	private function prepare_file() {
+		$log_file = $this->get_log_file();
+		if ( file_exists( $log_file ) ) {
+			return true;
+		}
+
+		$directory = dirname( $log_file );
+		if ( ! Boost_Cache_Utils::create_directory( $directory ) ) {
+			return new \WP_Error( 'Could not create boost cache log directory' );
+		}
+
+		return Boost_Cache_Utils::write_to_file( $log_file, self::LOG_HEADER );
+	}
+
+	/**
+	 * Add a debug message that logs request specific events.
+	 *
+	 * Use this for logging cache hits, misses, and other request specific events.
+	 * It will filter out requests for wp-admin and other non-cacheable requests.
+	 *
+	 * @param string $message - The message to write to the log file.
+	 */
+	public static function request_debug( $message ) {
+		$request = new Request();
+		if ( $request->is_cacheable() ) {
+			self::debug( $message );
+		}
+	}
```

Comment:
This needs cleaning up in a follow-up PR.

- Re-running the Request constructor re-normalizes the path, so it's unnecessarily heavy.
- The name and description of this function does not make it clear what it means/does.

Reply:
Do you think we should make the Request a `singleton`?